packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -54,15 +54,12 @@ export class LibraryDeploySourcePathExecutor extends DeployExecutor<
     response: ContinueResponse<string | string[]>
   ): Promise<ComponentSet> {
     const paths = response.data;
-    const components = new ComponentSet();
+
     if (typeof paths === 'string') {
-      components.resolveSourceComponents(paths);
+      return ComponentSet.fromSource(paths);
```

Comment:
Do we need to keep this if/else statements if both run `return ComponentSet.fromSource(paths);` ?

Reply:
This is probably my lack of understanding with typescript. When I leave it as `return ComponentSet.fromSource(paths);` I get this:

salesforcedx-vscode-core: src/commands/forceSourceDeploySourcePath.ts(58,36): error TS2769: No overload matches this call.
salesforcedx-vscode-core:   Overload 1 of 3, '(fsPath: string): ComponentSet', gave the following error.
salesforcedx-vscode-core:     Argument of type 'string | string[]' is not assignable to parameter of type 'string'.
salesforcedx-vscode-core:       Type 'string[]' is not assignable to type 'string'.
salesforcedx-vscode-core:   Overload 2 of 3, '(fsPaths: string[]): ComponentSet', gave the following error.
salesforcedx-vscode-core:     Argument of type 'string | string[]' is not assignable to parameter of type 'string[]'.
salesforcedx-vscode-core:       Type 'string' is not assignable to type 'string[]'.
salesforcedx-vscode-core:   Overload 3 of 3, '(options: FromSourceOptions): ComponentSet', gave the following error.
salesforcedx-vscode-core:     Argument of type 'string | string[]' is not assignable to parameter of type 'FromSourceOptions'.
salesforcedx-vscode-core:       Type 'string' is not assignable to type 'FromSourceOptions'.

Reply:
Yeah this is a funny Typescript thing, you have to either assert it's a `string` or `string[]`, not `string | string[]` type.

Reply:
@brpowell Is there a better way to add this assertion? Or is this approach sufficient?

Reply:
The other thing you could do is normalize it to an array:

```typescript
const fsPaths = typeof response.data === 'string' ? [response.data] : response.data
```

Not sure if there's a better way other than that.

Reply:
Thanks for confirming.